### DISCUSSION TITLE
2D scatter plot example

### DISCUSSION THREAD

**Discussion by rsand-temple** [jzy3d/jzy3d-api#245] [Q&A]
Hi - I am trying to make a simple 2D scatter plot of doubles. The method below takes in a 2d array of doubles with 1000 rows and 2 columns representing an array of cartesian coordinates. What I get is a blank chart except for a single red dot in the center. Any advice on how to get something meaningful? Thanks!

```
    public static void plotScatter2D(double[][] points, Color color) {
        EmulGLChartFactory factory = new EmulGLChartFactory();
        Quality q = Quality.Advanced();
        q.setAnimated(false); // leave CPU quiet if no need to re-render
        q.setHiDPIEnabled(true); // need java 9+ to enable HiDPI & Retina displays

        ScatterSerie2d scatter = new ScatterSerie2d("Scatter2D");
        scatter.setColor(color);
        scatter.setWidth(3);
        for (int i = 0; i < size; i++)
            scatter.add(points[i][0], points[i][1]);

        Chart2d chart = new Chart2d(factory, q);
        chart.add(scatter.getDrawable());

        chart.open();
        chart.addMouseCameraController();
    }
```

**Comment by jzy3d** [comment]
Hi, 

Sorry for missing this discussion. You can see a working example of a 2D scatter plot [here](https://github.com/jzy3d/jzy3d-api/blob/master/jzy3d-tutorials/src/main/java/org/jzy3d/demos/chart2d/Scatter2D_DemoAWT.java). It was added in may 2022 in version `2.0.1-SNAPSHOT`

Which generates
![Capture d’écran 2022-05-24 à 16 34 44](https://user-images.githubusercontent.com/2334960/170061713-eed268c3-49ed-4e21-94db-fcb83a07a61e.png)
